overlord/assertstate/assertmgr.go

```diff
@@ -119,17 +119,17 @@ func doValidateSnap(t *state.Task, _ *tomb.Tomb) error {
 	})
 	if notFound, ok := err.(*asserts.NotFoundError); ok {
 		if notFound.Type == asserts.SnapRevisionType {
-			return fmt.Errorf("cannot verify snap %q, no matching signatures found", snapsup.Name())
+			return fmt.Errorf("cannot verify snap %q, no matching signatures found", snapsup.InstanceName())
 		} else {
-			return fmt.Errorf("cannot find supported signatures to verify snap %q and its hash (%v)", snapsup.Name(), notFound)
+			return fmt.Errorf("cannot find supported signatures to verify snap %q and its hash (%v)", snapsup.InstanceName(), notFound)
 		}
 	}
 	if err != nil {
 		return err
 	}
 
 	db := DB(t.State())
-	err = snapasserts.CrossCheck(snapsup.Name(), sha3_384, snapSize, snapsup.SideInfo, db)
+	err = snapasserts.CrossCheck(snapsup.InstanceName(), sha3_384, snapSize, snapsup.SideInfo, db)
```

Comment:
that value will be compared with the snap-declaration snap-name,  so we need to pass SnapName (but then errors from that function will use it)  or we need to teach CrossCheck what to do about instance names

Reply:
@bboozzoo lets make sure we also have a test for this (either spread or unit) to make sure we never regress here.